### PR TITLE
github/workflows: Unified workflow, use buildah & podman for Arch & Ubuntu images

### PR DESCRIPTION
**Pull request by travier**
github/workflows/arch: Unified workflow, use buildah & podman

- Use a unified workflow for both PR & Push jobs
- Build using buildah & push with podman

---

github/workflows/ubuntu: Unified workflow, use buildah & podman

- Use a unified workflow for both PR & Push jobs
- Build using buildah & push with podman

### CODE REVIEW

**Comment by travier** [comment]
Imported from https://github.com/toolbx-images/images

**Review by Foxboron — APPROVED** [review]

File: .github/workflows/ubuntu-images.yaml [hunk 1]
```diff
@@ -10,52 +18,79 @@ on:
   schedule:
     - cron: '0 0 * * MON'
 
-# Prevent multiple workflow runs from racing
-concurrency: ${{ github.workflow }}
-
 env:
-  latest_release: '22.04'
+  distro: 'ubuntu'
+  latest_release: '23.10'
```
Comment (Jmennius) [inline .github/workflows/ubuntu-images.yaml:23]: My preference would be to leave this set to latest LTS :)

Comment (travier) [reply .github/workflows/ubuntu-images.yaml]: Good catch. Updated.

**Comment by travier** [comment]
@debarshiray Are we good to merge this one first? We'll decide later for the keys for https://github.com/containers/toolbox/pull/1440

**Comment by Foxboron** [comment]
I don't mind hitting merge if @Jmennius approves of the changes :)

**Review by Jmennius — APPROVED** [review]

**Comment by debarshiray** [comment]
> I don't mind hitting merge if @Jmennius approves of the changes :)

Thanks for getting this done, @travier @Foxboron @Jmennius 

I got a bit buried doing maintenance for the Fedora and RHEL images because of the impending port to [Kiwi](https://fedoraproject.org/wiki/Changes/KiwiBuiltCloudImages), and various internal deadlines and build system oddities.

**Pull request merged**